plugin-taskbar/lxqttaskgroup.cpp

```diff
@@ -631,66 +632,48 @@ bool LXQtTaskGroup::onWindowChanged(WId window, NET::Properties prop, NET::Prope
+        if (prop == LXQtTaskBarWindowProperty::Workspace)
         {
             if (parentTaskBar()->isShowOnlyOneDesktopTasks()
-                    || parentTaskBar()->isShowOnlyCurrentScreenTasks())
+                || parentTaskBar()->isShowOnlyCurrentScreenTasks())
             {
                 needsRefreshVisibility = true;
             }
         }
 
-        if (prop.testFlag(NET::WMVisibleName) || prop.testFlag(NET::WMName))
+        if (prop == LXQtTaskBarWindowProperty::Title)
             std::for_each(buttons.begin(), buttons.end(), std::mem_fn(&LXQtTaskButton::updateText));
 
         // XXX: we are setting window icon geometry -> don't need to handle NET::WMIconGeometry
         // Icon of the button can be based on windowClass
-        if (prop.testFlag(NET::WMIcon) || prop2.testFlag(NET::WM2WindowClass))
+        if (prop == LXQtTaskBarWindowProperty::Icon)
             std::for_each(buttons.begin(), buttons.end(), std::mem_fn(&LXQtTaskButton::updateIcon));
 
         bool set_urgency = false;
         bool urgency = false;
 
-        if (prop2.testFlag(NET::WM2Urgency))
+        if (prop == LXQtTaskBarWindowProperty::Urgency)
         {
             set_urgency = true;
-            if (auto *x11Application = qGuiApp->nativeInterface<QNativeInterface::QX11Application>())
-            {
-                WId appRootWindow = XDefaultRootWindow(x11Application->display());
-                urgency = NETWinInfo(x11Application->connection(), window, appRootWindow, NET::Properties{}, NET::WM2Urgency).urgency();
-            }
+            //FIXME: original code here did not consider "demand attention", was it intentional?
+            urgency = mBackend->applicationDemandsAttention(window);
         }
-        if (prop.testFlag(NET::WMState))
+        if (prop == LXQtTaskBarWindowProperty::State)
         {
-            KWindowInfo info{window, NET::WMState};
-
-            if(!set_urgency)
-            {
-                if (auto *x11Application = qGuiApp->nativeInterface<QNativeInterface::QX11Application>())
-                {
-                    WId appRootWindow = XDefaultRootWindow(x11Application->display());
-                    urgency = NETWinInfo(x11Application->connection(), window, appRootWindow, NET::Properties{}, NET::WM2Urgency).urgency();
-                }
-            }
-
-            // Force refresh urgency
-            //TODO: maybe do it in common place with NET::WM2Urgency
-            std::for_each(buttons.begin(), buttons.end(), std::bind(&LXQtTaskButton::setUrgencyHint, std::placeholders::_1, urgency || info.hasState(NET::DemandsAttention)));
+            if (!set_urgency)
+                urgency = mBackend->applicationDemandsAttention(window);
+            std::for_each(buttons.begin(), buttons.end(), std::bind(&LXQtTaskButton::setUrgencyHint, std::placeholders::_1, urgency));
             set_urgency = false;
 
-            if (info.hasState(NET::SkipTaskbar))
-                onWindowRemoved(window);
```

Comment:
Any idea why the above two lines were included in the first place?

Reply:
Well windows could change the "skip taskbar" property at any time, so the taskbar should remove/re-add them based on the value.
This logic was moved to backend, although it was only removing exiting non-acceptable window like this code does while it was missing the "re-add" part.
Fixed in last commit